nucypher/network/middleware.py

```diff
@@ -191,14 +190,7 @@ def propose_arrangement(self, node, arrangement):
         response = self.client.post(node_or_sprout=node,
                                     path="consider_arrangement",
                                     data=bytes(arrangement),
-                                    timeout=2)
-        return response
-
-    def enact_policy(self, ursula, kfrag_id, payload):
-        response = self.client.post(node_or_sprout=ursula,
-                                    path=f'kFrag/{kfrag_id.hex()}',
-                                    data=payload,
-                                    timeout=2)
+                                    timeout=120)  # TODO: What is an appropriate timeout here?
```

Comment:
Left from debugging?

Reply:
It feels that when the maximum wait is 2 minutes, some form of a progress bar is justified (or at the very least, a message for the user). I'd think the whole thing hung if I had to wait a minute.

Reply:
Opened https://github.com/nucypher/nucypher/issues/2738